osquery/core/plugins/logger.cpp

```diff
@@ -48,6 +48,8 @@ Status LoggerPlugin::call(const PluginRequest& request,
   std::vector<StatusLogLine> intermediate_logs;
   if (request.count("string") > 0) {
     return this->logString(request.at("string"));
+  } else if (request.count("string_batch") > 0) {
+    return this->logStringBatch(request.at("string_batch"));
```

Comment:
Why have both string and string_batch? It looks like they both take an array? Are there cases one would prefer to call string?

Reply:
> Why have both string and string_batch? It looks like they both take an array? Are there cases one would prefer to call string?

Only the **string_batch** request takes an array, which will then read and pass to the normal **string** request.

Using **string_batch** instead of **string** ensures that when you call this API from an extension you only issue a single Thrift request instead of one per entry in the array. Performance wise, this avoids a lot of overhead (protocol + 1 fork per API call within osquery)

Reply:
Ah, I misunderstood. So string takes a singleton, and string_batch takes an array. Okay then!